common/component/kafka/subscriber.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafka
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+// Subscribe adds a handler and configuration for a topic, and subscribes.
+// Unsubscribes to the topic on context cancel.
+func (k *Kafka) Subscribe(ctx context.Context, handlerConfig SubscriptionHandlerConfig, topics ...string) {
+	k.subscribeLock.Lock()
+	defer k.subscribeLock.Unlock()
+	for _, topic := range topics {
+		k.subscribeTopics[topic] = handlerConfig
```

Comment:
Should this check if there's already a value for `k.subscribeTopics`? Otherwise, we may be in a bad state, where one subscriber stopping causes the other subscriber to stop receiving messages (see the `delete` below)